File: src/com/dafttech/terra/game/world/entities/EntitySaplingArrow.java
package com.dafttech.terra.game.world.entities;

import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.World;
import com.dafttech.terra.game.world.tiles.Tile;
import com.dafttech.terra.game.world.tiles.TileSapling;

public class EntitySaplingArrow extends EntityFlamingArrow {

    public EntitySaplingArrow(Vector2 pos, World world) {
        super(pos, world);
    }

    @Override
    public void placeBlockOnHit(int x, int y) {
        Tile spreadTile = worldObj.getTile(x, y);
        if (spreadTile.isReplacable()) {
            worldObj.setTile(x, y, new TileSapling(), true);
        }
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/entities/EntitySaplingArrow.scala
package com.dafttech.terra.game.world.entities

import com.dafttech.terra.engine.TilePosition
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.tiles.TileSapling
import com.dafttech.terra.game.world.{GameWorld, TileInstance}

class EntitySaplingArrow(pos: Vector2d)(implicit world: GameWorld) extends EntityFlamingArrow(pos) {
  override def placeBlockOnHit(p: TilePosition): Unit = {
    val spreadTile = p.world.getTile(p.pos)
    if (spreadTile.tile.isReplacable) p.world.setTile(p.pos, TileInstance(new TileSapling()), notify = true)
  }
}